### ISSUE TITLE
if and opening bracket without a space

### ISSUE THREAD

**Issue by jayvdb** [PyCQA/pycodestyle#570]
Expanding on https://github.com/PyCQA/pycodestyle/issues/371#issuecomment-137404784 by @methane, the following are not flagged as erroneous:

``` python
if(1 == 2):
    pass

if[1, 2]:
    pass

if{1, 2}:
    pass

for(a, b) in [(1, 2), (3, 4)]:
    pass
```

pep8 doesnt appear to clearly reject the first two (parentheses), however it goes against the recommendation of `if (` in the line continuation section.

i.e. the following is not appropriate according to pep8, but it doesnt use extremely strong wording regarding this.

``` python
if(foo or
        bar):
    pass
```

pep8 says:
"When the conditional part of an `if` -statement is long enough to require that it be written across multiple lines, it's worth noting that the combination of a two character keyword (i.e. `if` ), plus a single space, plus an opening parenthesis creates a natural 4-space indent for the subsequent lines of the multiline conditional.  This can produce a visual conflict with the indented suite of code nested inside the if -statement, which would also naturally be indented to 4 spaces. This PEP takes no explicit position on how (or whether) to further visually distinguish such conditional lines from the nested suite inside the if -statement.  Acceptable options in this situation include, but are not limited to:" _(and then it includes examples all containing_ `if (` _.)_

The `further` there indicates that the `if (` is expected, and of course the examples throughout all use `if (`.
Ironically pycodestyle does take an explicit position on further visually distinguishing subsequent lines. 

For the unassigned data-structures, there is less in pep8 regarding whether they should be surrounded by space.  A single space is required around `=`, `:`, and `,`, and prevented when immediately inside a function call, so that addresses most cases of unassigned data-structures.  But it doesnt address when unassigned data-structures are used immediately near a keyword like `if` or `for`.

I suspect some revision to the pep8 style guide is appropriate, but the fine people here can surely help bring other perspectives to this aspect of the style guide before trying to refine the style guide.

**Comment by asottile** [comment]
Duplicate of #371
